stdlib/public/runtime/Exclusivity.cpp

```diff
@@ -202,6 +202,7 @@ class AccessSet {
   }
 
   void remove(Access *access) {
```

Comment:
How does this new assert help? If the access set is empty, you'll always hit the assert below.

Reply:
If it's empty, you'll trigger a seg-fault on the `cur->getNext()` (the main caller doesn't call this method if `access` is `nullptr`). When I was debugging this with a Release+Asserts compiler, it doesn't directly point to that line, since it gets inlined / optimized somewhat. That's why I added it. 

Do you think I should remove this assert?

Reply:
I see. I think the assert is good, just didn't understand that it was avoiding a non-assert crash.